jetty-websocket/websocket-javax-common/src/main/java/org/eclipse/jetty/websocket/javax/common/JavaxWebSocketFrameHandlerFactory.java

```diff
@@ -165,13 +165,16 @@ public JavaxWebSocketFrameHandler newJavaxWebSocketFrameHandler(Object endpointI
         errorHandle = InvokerUtils.bindTo(errorHandle, endpoint);
         pongHandle = InvokerUtils.bindTo(pongHandle, endpoint);
 
-        return new JavaxWebSocketFrameHandler(
+        JavaxWebSocketFrameHandler frameHandler = new JavaxWebSocketFrameHandler(
             container,
             endpoint,
             openHandle, closeHandle, errorHandle,
             textMetadata, binaryMetadata,
             pongHandle,
             config);
+
+        frameHandler.setUpgradeRequest(upgradeRequest);
```

Comment:
Why isn't this done with constructor injection?

Reply:
It was because that is how it is done on the Jetty WebSocket side, we set these in `WebSocketNegotiator.negotiate()` so that we have the response available as well, but the response isn't needed on the javax side so we can add this right after we create it.

I have now removed this `setUpgradeRequest()` method, and add the upgrade request in the constructor.